apps/auth/src/middleware.ts

```diff
@@ -97,18 +120,17 @@ export async function middleware(request: NextRequest) {
       return new NextResponse('Invalid redirect URL', { status: 400 })
     }
 
-    console.log('Valid redirect URL, proceeding with OAuth redirect')
-
-    // Redirect directly to the EduID OAuth flow
-    const signinUrl = new URL('/api/auth/signin/eduid-participant', request.url)
-    signinUrl.searchParams.set('callbackUrl', redirectTo)
-    signinUrl.searchParams.set('participant', 'true')
-
-    console.log('Student route redirect to:', signinUrl.toString())
-
-    const response = NextResponse.redirect(signinUrl)
-
-    console.log('Returning redirect response')
+    // Set/refresh the redirect cookie so it's available on callback even if
+    // NextAuth posts the callbackUrl in the body (not readable in middleware)
+    const response = NextResponse.next()
+    response.cookies.set(REDIRECT_COOKIE_NAME, redirectTo, {
+      httpOnly: true,
+      sameSite: 'lax',
+      secure: process.env.NODE_ENV === 'production',
+      path: '/',
+      maxAge: 60 * 10, // 10 minutes to account for user delays
```

Comment:
This max age is not 10 minutes, but 600 milliseconds - this might affect slower redirects already? 😅